twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,322 @@
+ angular.module('cloudberry.cache', ['leaflet-directive', 'cloudberry.common' ])
+ .service('Cache', function( $window, $http, $compile){
+
+  var cachedCityPolygonTree = rbush();
+  var cachedRegion ;
+  var cacheSize = 0;
+  var insertedTreeIDs = new Set();
+  var cacheThreshold = 7500;//hard limit
+  var DeleteTarget = 0;
+  var DeletedCount = 0;
+  var preFetchDistance = 25;
+
+
+  /*Call will happen from map controller and this functions sees whether a required data is present in cache or not
+  if not gets from midleware*/
+  this.getCityPolygonsFromCache = function city(bounds){
+
+       var deferred = new $.Deferred();
+       var data_response;
+
+
+       var rteBounds = "city/" + bounds._northEast.lat + "/" + bounds._southWest.lat + "/" + bounds._northEast.lng + "/" + bounds._southWest.lng;
+
+       var currentRequestPolygon = turf.polygon([[
+                                    [bounds._northEast.lng,bounds._northEast.lat],
+                                    [bounds._northEast.lng,bounds._southWest.lat],
+                                    [bounds._southWest.lng,bounds._southWest.lat],
+                                    [bounds._southWest.lng,bounds._northEast.lat],
+                                    [bounds._northEast.lng,bounds._northEast.lat]
+                                ]]);
+
+       var bbox = turf.bbox(currentRequestPolygon);
+        // to search in Rbush Tree we need this in this format
+       var item = {
+           minX : bbox[0],
+           minY : bbox[1],
+           maxX : bbox[2],
+           maxY : bbox[3]
+       }
+
+
+       if(typeof cachedRegion != "undefined" && typeof turf.difference(currentRequestPolygon,cachedRegion) == "undefined") {
+              //cache HIT
```

Comment:
I think @JavierJia also mentioned about this. @vidhya567 : can you check his comments on the previous commit and apply them?